proof/invariant-abstract/KHeap_AI.thy

```diff
@@ -2441,8 +2441,9 @@ lemma set_tcb_obj_ref_obj_at_trivial:
   by (wpsimp wp: set_tcb_obj_ref_wp simp: obj_at_def)
 
 lemma sched_context_update_consumed_obj_at_trivial:
-  "\<lbrace>obj_at P t' and
-    K (\<forall>sc n. P (SchedContext sc n) \<longrightarrow> P (SchedContext (sc_consumed_update (K 0) sc) n))\<rbrace>
+  "\<lbrace>obj_at P t'
```

Comment:
I found in my PR that if you add a "forall f" as well, then you can have an arbitrary f to update the consumed time and everything goes through just fine

Reply:
Awesome!